src/main/java/org/broadinstitute/hellbender/tools/sv/stratify/SVStatificationEngine.java

```diff
@@ -0,0 +1,347 @@
+import org.broadinstitute.hellbender.utils.IntervalMergingRule;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableUtils;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+// Groups variants by SVTYPE, SVLEN, and overlap with one or more interval sets
+public class SVStatificationEngine {
+
+    // Configuration table column names
+    public static final String NAME_COLUMN = "NAME";
+    public static final String SVTYPE_COLUMN = "SVTYPE";
+    public static final String MIN_SIZE_COLUMN = "MIN_SIZE";
+    public static final String MAX_SIZE_COLUMN = "MAX_SIZE";
+    public static final String CONTEXT_COLUMN = "CONTEXT";
+    protected static final Set<String> COLUMN_NAMES = ImmutableSet.of(NAME_COLUMN, SVTYPE_COLUMN, MIN_SIZE_COLUMN, MAX_SIZE_COLUMN, CONTEXT_COLUMN);
+    public static final String CONTEXT_COLUMN_DELIMITER = ",";
+
+    public static final Set<String> NULL_TABLE_VALUES = Set.of("-1", "", "NULL", "NA");
+
+    protected final Map<String, OverlapDetector<Locatable>> contextMap;
+    protected final Map<String, Stratum> strata;
+    protected final SAMSequenceDictionary dictionary;
+
+    public SVStatificationEngine(final SAMSequenceDictionary dictionary) {
+        contextMap = new HashMap<>();
+        strata = new HashMap<>();
+        this.dictionary = Utils.nonNull(dictionary);
+    }
+
+    public void addContext(final String name, final List<Locatable> intervals) {
+        Utils.nonNull(name);
+        Utils.nonNull(intervals);
+        Utils.validateArg(!contextMap.containsKey(name), "Context with name " + name + " already exists");
+        contextMap.put(name, OverlapDetector.create(intervals));
+    }
+
+    /**
+     * Adds a new stratification group
+     * @param name a unique ID
+     * @param svType SV type, may be null
+     * @param minSize minimum size in bp (inclusive), may be null
+     * @param maxSize maximum size in bp (exclusive), may be null
+     * @param contextNames reference context names
+     */
+    public void addStratification(final String name, final GATKSVVCFConstants.StructuralVariantAnnotationType svType,
+                                  final Integer minSize, final Integer maxSize, final Set<String> contextNames) {
+        addStratification(new Stratum(name, svType, minSize, maxSize, contextNames));
+    }
+
+    protected void addStratification(final Stratum stratification) {
+        Utils.validateArg(!strata.containsKey(stratification.getName()), "Encountered duplicate name " + stratification.getName());
```

Comment:
Can you wrap this in a UserException?  I've been burned checking for IllegalArgumentExceptions in tests (when another class throws an IllegalArgumentException)

Reply:
How about a `UserException`? Since this is a backend class